src/Sarif/EnumeratedArtifact.cs

```diff
@@ -87,50 +91,28 @@ public byte[] Bytes
                 this.Stream = FileSystem.FileOpenRead(Uri.LocalPath);
             }
 
-            if (Stream.CanSeek)
-            {
-                RetrieveDataFromSeekableStream();
-            }
-            else
-            {
-                RetrieveDataFromNonSeekableStream();
-            }
+            RetrieveDataFromStream();
 
             this.Stream = null;
 
             return (this.contents, this.bytes);
         }
 
-        private void RetrieveDataFromNonSeekableStream()
+        private void RetrieveDataFromStream()
```

Comment:
>RetrieveDataFromStream [](http://example.com/codeflow?start=21&length=22)

Nice abstraction you've delivered! Look how clean and readable this code is now.